asv/commands/run.py

```diff
@@ -238,7 +238,8 @@ def run(cls, conf, range_spec=None, steps=None, date_period=None,
 
         if range_spec is None:
             try:
-                commit_hashes = list(set([repo.get_hash_from_name(branch) for branch in conf.branches]))
+                commit_hashes = list(set([(repo.get_hash_from_name(branch) for
```

Comment:
You are adding a parenthesis after the squared bracket. That is changing the code, instead of the comprehension returning a list of the iterated items, this is returning a list with a single value which is a generator. This is why tests are failing.

Reply:
Thank you for the explanation. I'm going to fix it!